staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -121,17 +145,14 @@ func (a *mutatingDispatcher) Dispatch(ctx context.Context, attr admission.Attrib
 			continue
 		}
 
-		if versionedAttr == nil {
-			// First webhook, create versioned attributes
-			var err error
-			if versionedAttr, err = admission.NewVersionedAttributes(attr, invocation.Kind, o); err != nil {
-				return apierrors.NewInternalError(err)
-			}
-		} else {
-			// Subsequent webhook, convert existing versioned attributes to this webhook's version
-			if err := admission.ConvertVersionedAttributes(versionedAttr, invocation.Kind, o); err != nil {
-				return apierrors.NewInternalError(err)
-			}
+		// Subsequent webhook, convert existing versioned attributes to this webhook's version
+		var err error
+		versionedAttr, err = v.VersionedAttribute(attr, o, invocation.Kind)
+		if err != nil {
+			return apierrors.NewInternalError(err)
+		}
+		if err = admission.ConvertVersionedAttributes(versionedAttr, invocation.Kind, o); err != nil {
+			return apierrors.NewInternalError(err)
 		}
```

Comment:
I expected this if/else block to move inside the accessor, like this (trimming down the interface to just take `gvk` and baking `attr admission.Attributes, o admission.ObjectInterfaces` into the accessor):

```go
func (v *versionedAttributeAccessor) VersionedAttributes(gvk schema.GroupVersionKind) (*admission.VersionedAttributes, error) {
	if v.versionedAttr == nil {
		// First call, create versioned attributes
		var err error
		if v.versionedAttr, err = admission.NewVersionedAttributes(v.attr, gvk, v.o); err != nil {
			return nil, apierrors.NewInternalError(err)
		}
	} else {
		// Subsequent call, convert existing versioned attributes to the requested version
		if err := admission.ConvertVersionedAttributes(v.versionedAttr, gvk, v.o); err != nil {
			return nil, apierrors.NewInternalError(err)
		}
	}
	return v.versionedAttr, nil
}
```

for the call here to become:
```
		versionedAttr, err := v.VersionedAttribute(invocation.Kind)
		if err != nil {
			return err
		}
```

and to only access the versionedAttr held by the accessor below, like this:

```go
	// convert versionedAttr.VersionedObject to the internal version in the underlying admission.Attributes
	if v.versionedAttr != nil && v.versionedAttr.VersionedObject != nil && v.versionedAttr.Dirty {
		return o.GetObjectConvertor().Convert(v.versionedAttr.VersionedObject, v.versionedAttr.Attributes.GetObject(), nil)
	}
```

Reply:
ah nice, will update, thank you!